### PR TITLE
cherry-pick & reolved

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 1]
```diff
@@ -269,7 +269,7 @@
                             <td class="select2-parent pdR1">
                                 <div class="form-group">
                                     <label class="default_clr">Voucher Type</label>
-                                    <sh-select #sh [options]="voucherTypeList | async" [isFilterEnabled]="true"
+                                    <sh-select id="ledger-vouchar-type" #sh [options]="voucherTypeList | async" [isFilterEnabled]="true"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:272]: Incorrect spelling of voucher

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 2]
```diff
@@ -317,7 +317,7 @@
                                 <div class="form-group">
                                     <label class="default_clr">Voucher No.</label>
 
-                                    <sh-select [options]="invoiceList" #invoice placeholder="Select Invoice"
+                                    <sh-select id="ledger-vouchar-number" [options]="invoiceList" #invoice placeholder="Select Invoice"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:320]: Incorrect spelling of voucher

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 3]
```diff
@@ -526,22 +524,22 @@
...
                     </button>
 
-                    <button [hidden]="isBankTransaction" class="btn btn-default btn-sm"
+                    <button id="ledger-add-button-cr" [hidden]="isBankTransaction" class="btn btn-default btn-sm"
                         (click)="addToDrOrCr('CREDIT',$event)">Add
                         to CR
                     </button>
 
-                    <button id="saveLedger" class="btn btn-success btn-sm mr-0" [disabled]="isFileUploading"
+                    <button id="ledger-save-button" id="saveLedger" class="btn btn-success btn-sm mr-0" [disabled]="isFileUploading"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:542]: Avoid adding two IDs to an element

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 4]
```diff
@@ -384,13 +384,13 @@
...
                                     <label class="default_clr">Cheque Number</label>
-                                    <input type="text" placeholder="XXXX2619" class="form-control"
+                                    <input id="ledger-cheque-number" type="text" placeholder="XXXX2619" class="form-control"
                                         [(ngModel)]="blankLedger.chequeNumber" />
                                 </td>
 
                                 <td>
                                     <label class="default_clr">Cheque Clearance Date</label>
-                                    <input type="text" autocomplete="off" class="form-control" id="chequeClearanceDate"
+                                    <input id="ledger-cheque-clearance" type="text" autocomplete="off" class="form-control" id="chequeClearanceDate"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:393]: Avoid adding two IDs to an element

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 5]
```diff
@@ -758,27 +758,21 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
...
+                                        <input id="update-ledger-inclusive-tax" type="text" class="form-control mb-1" decimalDigitsDirective [mask]="vm.inputMaskFormat" [prefix]="vm.selectedPrefixForCurrency" (blur)="adjustedInvoiceAmountChange()" *ngIf="selectedAdvanceReceiptAdjustInvoiceEditMode" [(ngModel)]="item.adjustmentAmount.amountForAccount">
                                     </div>
                                 </div>
                                 <div *ngIf="item.voucherType === 'receipt'" class="col-sm-2 pl-05">
                                     <div class="form-group">
                                         <label>Tax</label>
-                                        <input type="text" class="form-control mb-1" decimalDigitsDirective [mask]="vm.inputMaskFormat"
-                                            [prefix]="vm.selectedPrefixForCurrency" [(ngModel)]="item.calculatedTaxAmount" disabled>
+                                        <input type="text" id="update-ledger-vouchar-tax" class="form-control mb-1" decimalDigitsDirective [mask]="vm.inputMaskFormat" [prefix]="vm.selectedPrefixForCurrency" [(ngModel)]="item.calculatedTaxAmount" disabled>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:775]: Incorrect spelling of voucher

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 6]
```diff
@@ -876,7 +870,7 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
                                 <td class="select2-parent pdR1">
                                     <div class="form-group">
                                         <label class="default_clr">Voucher Type</label>
-                                        <sh-select [options]="vm.voucherTypeList" [isFilterEnabled]="true"
+                                        <sh-select id="update-ledger-vouchar-type" [options]="vm.voucherTypeList" [isFilterEnabled]="true"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:873]: Incorrect spelling of voucher

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 7]
```diff
@@ -924,7 +918,7 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
                                     <div class="form-group">
                                         <label class="default_clr">{{vm.selectedLedger.voucher.shortCode == 'debit note' ? 'Voucher' : 'Invoice'}} No.</label>
 
-                                        <sh-select [options]="invoiceList" placeholder="Select Accounts"
+                                        <sh-select id="update-ledger-invoice-number" [options]="invoiceList" placeholder="Select Accounts"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:921]: It has same ID as element on line no. 885 please assign unique IDs

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 8]
```diff
@@ -951,7 +945,7 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
                                 <td class="pdR1">
                                     <div class="form-group">
                                         <label class="default_clr">Voucher No.</label>
-                                        <input type="text" placeholder="Voucher no." class="form-control"
+                                        <input id="update-ledger-vouchar-number" type="text" placeholder="Voucher no." class="form-control"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:948]: Incorrect spelling of voucher

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 9]
```diff
@@ -1094,7 +1088,7 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
                                 </div>
 
                                 <label class="upload_div" [ngClass]="{hide:vm.selectedLedger.attachedFile}">
-                                    <input type="file" name="invoiceFile" #fileInputUpdate id="invoiceFile" ngFileSelect
+                                    <input id="update-ledger-upload-file" type="file" name="invoiceFile" #fileInputUpdate id="invoiceFile" ngFileSelect
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:1091]: Avoid adding two IDs to an element

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 10]
```diff
@@ -59,37 +59,37 @@
...
-                    <div class="ledger-datepicker" container="body" *ngIf="isMobileScreen">
+                    <div class="ledger-datepicker" id="giddh-datepicker"  container="body" *ngIf="isMobileScreen">
                         <img (click)="showGiddhDatepicker($event)" src="{{imgPath}}calender-icon.svg">
                     </div>
                     <span class="input-wrapper ipad-view-datepicker">
                         <!-- <input type="text" class="form-control daterange_picker_ledger" daterangepicker
                             [options]="datePickerOptions" (applyDaterangepicker)="selectedDate($event)"
                             name="daterangeInput" /> -->
-                        <input type="text" name="selectedDateRange" (click)="showGiddhDatepicker($event)"
+                        <input id="giddk-datepicker" type="text" name="selectedDateRange" (click)="showGiddhDatepicker($event)"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:69]: Correct spelling of Giddh

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 11]
```diff
@@ -108,7 +108,7 @@
...
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:111]: Correct spelling of Giddh

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 12]
```diff
@@ -161,10 +161,10 @@
 
             <div class="toggle d-inline-block currencyToggler" data-key="disabled">
 
-                <input type="checkbox" class="toggle-checkbox" id="disabled" [(ngModel)]="currencyTogglerModel"
+                <input type="checkbox" id="currency-toggle" class="toggle-checkbox" id="disabled" [(ngModel)]="currencyTogglerModel"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:164]: Avoid adding two IDs to an element

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 13]
```diff
@@ -382,7 +382,7 @@ <h3 class="ledgerHead">
                                                                 <div class="d-flex align-items-center"
                                                                     >
                                                                     <span class="pl-05">
-                                                                        <input type="checkbox"
+                                                                        <input type="checkbox" id="select-entry-cr"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:385]: ID provided is same as the element on line 270 and several other places below this code in the same file. Please make sure the elements have unique IDs

File: package-lock.json [hunk 14]
```diff
@@ -5313,7 +5313,7 @@
       }
     },
     "electron-oauth": {
-      "version": "git+https://github.com/govi2010/electron-oauth2.git#895adf1eadc2421b77ad0209eae9bec93850af3b",
```
Comment (prateek-giddh) [inline package-lock.json:5316]: Please revert this file

**Review by prateek-giddh — CHANGES_REQUESTED** [review]
Please check if all the elements have unique IDs and check elements have multiple ID attributes, remove cautiously as the old IDs might have styles in CSS files

File: package-lock.json (hunk 14 above)
Comment (shalinee-walkover) [reply package-lock.json]: Reverted

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html (hunk 1 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html]: All changes are done Please check

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**